internal/cmd/commands/server/server.go

```diff
@@ -630,3 +633,23 @@ func (c *Command) Reload() error {
 
 	return reloadErrors.ErrorOrNil()
 }
+
+func (c *Command) verifyKmsSetup() error {
+	const op = "server.(Command).verifyKmsExists"
+	rw := db.New(c.Database)
+
+	kmsRepo, err := kms.NewRepository(rw, rw)
+	if err != nil {
+		return fmt.Errorf("error creating kms repository: %w", err)
+	}
+	rks, err := kmsRepo.ListRootKeys(c.Context, kms.WithLimit(1))
+	if err != nil {
+		return err
+	}
+	for _, rk := range rks {
+		if rk.GetScopeId() == scope.Global.String() {
```

Comment:
@jimlambrt Do you think we should list scopes as well, and then verify that we have a root key for each scope? It could also be a future enhancement. At some point we could get all the way down the rabbit hole and verify that for each known type of key we have a key for each scope, but I don't know where the point of diminishing returns is here.

Reply:
For now, I think this is probably sufficient.  A scope's creation along with it's associated root key and deks are all in one transaction.  I don't think, the API provides anyway for an operator to delete keys on their own... so at this point, I don't think the db can get into a state where there's a scope without it's root and deks.